src/python_bindings/bindings.cpp

```diff
@@ -125,15 +123,30 @@ class PyHNSWLibIndex : public PyVecSimIndex {
         hnsw->setEf(ef);
     }
     void saveIndex(const std::string &location) {
-        auto serializer = hnswlib::HNSWIndexSerializer<float>(
-            reinterpret_cast<HNSWIndex *>(index)->getHNSWIndex());
-        serializer.saveIndex(location);
+        if (vecType == VecSimType_FLOAT32) {
+            if (metric == VecSimMetric_Cosine || metric == VecSimMetric_IP ||
+                metric == VecSimMetric_L2) {
+                auto serializer = hnswlib::HNSWIndexSerializer<float>(
+                    reinterpret_cast<HNSWIndex *>(index)->getHNSWIndex());
+                serializer.saveIndex(location);
+            }
+        }
     }
+
     void loadIndex(const std::string &location) {
-        auto serializer = hnswlib::HNSWIndexSerializer<float>(
-            reinterpret_cast<HNSWIndex *>(index)->getHNSWIndex());
-        serializer.loadIndex(location);
+        if (vecType == VecSimType_FLOAT32) {
+            if (metric == VecSimMetric_Cosine || metric == VecSimMetric_IP ||
+                metric == VecSimMetric_L2) {
+                auto serializer = hnswlib::HNSWIndexSerializer<float>(
+                    reinterpret_cast<HNSWIndex *>(index)->getHNSWIndex());
+                serializer.loadIndex(location);
+            }
+        }
     }
+
+private:
+    VecSimType vecType;
+    VecSimMetric metric;
```

Comment:
I don't think that these fields are necessary... you can use `VecSimIndex_Info(this->index)` instead to get the index type and create the appropriate serializer

Reply:
isn't it a bit redundant to call this function, and retrieve type and metric from the vecsiminfo struct, at the beginning of every function in this class (except the first)

Reply:
As I see it, the `vecType` field is already stored in `this->index`, so storing it again in another field of the class is a bit redundant. Though it's not critical...

Reply:
K